### PR TITLE
Make e2e specs travis friendly

### PR DESCRIPTION
**Pull request by gisikw**
Replaces #4974 
# End-to-End Specs Now Run in Travis
- Used copy/id fixes from #4974 as a starting point.
- Updated protractor version to 2.0.0, which includes a new version of the selenium-standalone jar (the existing version was incompatible with current versions of Firefox).
- Rewrote our test script using Coffeescript and ShellJS. Our existing setup required too much branching, and the trapping logic was still leaving some selenium/node processes orphaned. With this setup, selenium, node server, and xvfb are all child processes, so `Ctrl+c` or failures at any point during the testing suite will kill all the things.
- Added a summary to the end of the test results, to improve visibility of precisely _what_ is failing/passing. Each test suite will be run - regardless of whether its parents crash, so this summary should give a good indication of where to scroll to.

screenshot 2015-04-04 22 00 55

Unfortunately, had to comment out one of the e2e specs - the current combination of selenium/firefox run into an issue after clicking the "register" tab, whereby not all fields get filled in. I've been banging my head against a wall all day trying to get it working, and I believe it's related to [this Selenium issue](https://code.google.com/p/selenium/issues/detail?id=8180&thanks=8180&ts=1416358853). However, the `.clear()` and `.click()` fixes do not seem to apply (nor does adding timeouts, browser sleeps, structuring the entire thing as promises, nor praying to the Selenium gods, nor screaming very loudly at my laptop). Hopefully though, we can get this one spec up-and-running properly in a future PR :)

Anyway, Travis will now track the E2E specs! \o/

### CODE REVIEW

File: test/runTests.coffee [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    sh.echo ""
+    announce "Tests complete!\n\nSummary\n-------\n"
+    for name, result of results
+      if result is 0
+        sh.echo '\x1b[36m%s\x1b[0m', "#{name}: \x1b[32mpassing"
+      else
+        anyFailed = 1
+        sh.echo '\x1b[36m%s\x1b[0m', "#{name}: \x1b[31mfailing"
+    sh.echo ""
+    announce "Thanks for helping keep Habitica clean!"
```
Comment (crookedneighbor) [inline test/runTests.coffee:37]: Especially pleased to see this.

**Comment by crookedneighbor** [comment]
Love this. 

@nicktomlin, Travis will now run the e2e tasks. Thought you might be interested in seeing the set up.

**Comment by crookedneighbor** [comment]
I should be back up and running by the end of the week. If no one has merged this in by then, I'll test it out and merge it. Thanks for your patience on this! Very grateful for all the work you've been doing to improve the tests.

**Comment by Alys** [comment]
I assume this has to be tested in an environment where Firefox can be run in a GUI - right?

**Comment by gisikw** [comment]
You should be able to run it headless, so long as you have xvfb, firefox, and a Java runtime (for Selenium) installed. Travis is a headless env, and it's able to run the Firefox specs :)

**Comment by crookedneighbor** [comment]
What @gisikw said ^ :smile:

**Comment by crookedneighbor** [comment]
I've notated this pr towards your next tier.

**Comment by gisikw** [comment]
Cheers!

**Pull request merged**